.github/workflows/main.yml

```diff
@@ -38,6 +42,13 @@ jobs:
         scala_version: [ 2.12.17 ]
         os: [ ubuntu-20.04, windows-2019 ]
         include:
+          - os: macos-12
+            shell: bash
+            distribution: temurin
+            java_version: 17
+            scala_version: 2.12.17
+            env_cc: cc
+            env_ar: ar
```

Comment:
FYI, the reason why we used include was because we couldn't add macos-12 to `os: [ ubuntu-20.04, windows-2019 ]` without also getting `java_version: [ 8, 11, 17 ]` combinations as well, and we found out the rest of the variables would be empty if we didn't define them in include as well.

However, I've just noticed that we can use [exclude](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrixexclude) which could be more concise (we'd need only 4 exclude lines instead of 7 include lines, although we'd have to define env_cc, env_ar, shell in the matrix itself too).  I guess it's a matter of taste, but here goes anyway:

```yaml
      matrix:
        os: [ macos-12, ubuntu-20.04, windows-2019 ]
        distribution: [ temurin ]
        java_version: [ 8, 11, 17 ]
        env_cc: [ clang ]
        env_ar: [ ar ]
        scala_version: [ 2.12.17 ]
        shell: [ bash ]
        exclude:
          - os: macos-12
            java_version: 8
          - os: macos-12
            java_version: 11
        include:
          - os: ubuntu-20.04
            env_ar: llvm-ar-10
          - os: windows-2019
            env_ar: llvm-ar
            shell: msys2 {0}
```

We can go with the above if it looks better; it does seem to be more concise.